internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -311,9 +323,27 @@ func (h *PolicyChangeHandler) handlePolicyChange(ctx context.Context, c *config.
 		return fmt.Errorf("applying FleetClientConfig: %w", err)
 	}
 
+	if loggingHasChanged {
+		// The only way to update the logging configuration
+		// is to re-start the Elastic-Agent
+		h.coordinator.ReExec(nil)
+	}
+
 	return nil
 }
 
+// eventLoggingHasChanged returns true if the output of the event logger has changed
+func (p *PolicyChangeHandler) eventLoggingHasChanged(new *configuration.Configuration) bool {
+	switch {
+	case p.config.Settings.EventLoggingConfig.ToFiles != new.Settings.EventLoggingConfig.ToFiles:
+		return true
+	case p.config.Settings.EventLoggingConfig.ToStderr != new.Settings.EventLoggingConfig.ToStderr:
+		return true
+	default:
+		return false
+	}
+}
+
```

Comment:
```suggestion
func (h *PolicyChangeHandler) eventLoggingHasChanged(new *configuration.Configuration) bool {
	switch {
	case h.config.Settings.EventLoggingConfig.ToFiles != new.Settings.EventLoggingConfig.ToFiles:
		return true
	case h.config.Settings.EventLoggingConfig.ToStderr != new.Settings.EventLoggingConfig.ToStderr:
		return true
	default:
		return false
	}
}

```

to make lint happy, easy change (should be done)